packages/expect/src/index.ts

```diff
@@ -53,18 +52,18 @@ class JestAssertionError extends Error {
   matcherResult: any;
 }
 
-const isPromise = obj =>
+const isPromise = (obj: any) =>
   !!obj &&
   (typeof obj === 'object' || typeof obj === 'function') &&
   typeof obj.then === 'function';
 
-const createToThrowErrorMatchingSnapshotMatcher = function(matcher) {
-  return function(received: any, testNameOrInlineSnapshot?: string) {
+const createToThrowErrorMatchingSnapshotMatcher = function(matcher: any) {
+  return function(this: any, received: any, testNameOrInlineSnapshot?: string) {
```

Comment:
Not sure but should this be `MatcherState` here as well?

Reply:
Yes, `this` looks like `MatcherState`

If you need a type for `matcher` in the preceding line, it looks like `RawMatcherFn` currently exported from `types/Matchers.js`

Reply:
this has been updated